lib/stanford-mods/origin_info.rb

```diff
@@ -0,0 +1,223 @@
+require 'logger'
+require 'mods'
+
+# Parsing MODS /originInfo for Publication/Imprint data:
+#  * pub date for date slider facet
+#  * pub date for sort
+#  * imprint info for display
+#  *
+# These methods may be used by searchworks.rb file or by downstream apps
+module Stanford
+  module Mods
+    class Record < ::Mods::Record
+
+
+# -- likely to be private or protected
+
+      # return all /originInfo/dateCreated elements in MODS records
+      # @return [Nokogiri::XML::NodeSet<Nokogiri::XML::Element>]
+      def date_created_nodeset
+        @mods_ng_xml.origin_info.dateCreated   # returns nokogiri element/node objects
+      end
+
+      # return all /originInfo/dateIssued elements in MODS records
+      # @return [Nokogiri::XML::NodeSet<Nokogiri::XML::Element>]
+      def date_issued_nodeset
+        @mods_ng_xml.origin_info.dateIssued   # returns nokogiri element/node objects
+      end
+
+      # given a set of date elements, return the single element with attribute keyDate="yes"
+      #  or return nil if no elements have attribute keyDate="yes", or if multiple elements have keyDate="yes"
+      # @param [Nokogiri::XML::NodeSet<Nokogiri::XML::Element>] nodeset set of date elements
+      # @return [Nokogiri::XML::Element, nil] single date element with attribute keyDate="yes", or nil
+      def self.keyDate(nodeset)
+        keyDates = nodeset.select { |node| node["keyDate"] == 'yes' }
+        return keyDates.first if keyDates.size == 1
+      end
+
+      # @param [Nokogiri::XML::NodeSet<Nokogiri::XML::Element>] nodeset set of date elements
+      # @return [Nokogiri::XML::NodeSet<Nokogiri::XML::Element>] the set of date elements minus any that
+      #   had a qualifier attribute of 'approximate' or 'questionable'
+      def self.remove_approximate(nodeset)
+        nodeset.select { |node| node unless date_is_approximate?(node) }
+      end
+
+
+      # NOTE: legal values for MODS date elements with attribute qualifier are
+      #   'approximate', 'inferred' or 'questionable'
+      # @param [Nokogiri::XML::Element] date_element MODS date element
+      # @return [Boolean] true if date_element has a qualifier attribute of "approximate" or "questionable",
+      #   false if no qualifier attribute, or if attribute is 'inferred' or some other value
+      def self.date_is_approximate?(date_element)
+        qualifier = date_element["qualifier"] if date_element.respond_to?('[]')
+        qualifier == 'approximate' || qualifier == 'questionable'
+      end
+
+
+# ----   old
```

Comment:
What does "old" mean here? Are we going to throw these methods away, or are they here for compatibility, or?